rust/worker/src/tracing/opentelemetry_config.rs

```diff
@@ -1,22 +1,66 @@
 use opentelemetry::global;
-use opentelemetry::sdk::propagation::TraceContextPropagator;
-use opentelemetry::sdk::trace;
+use opentelemetry::trace::TracerProvider;
 use opentelemetry_otlp::WithExportConfig;
+use opentelemetry_sdk::propagation::TraceContextPropagator;
 use tracing_bunyan_formatter::BunyanFormattingLayer;
 use tracing_subscriber::{layer::SubscriberExt, EnvFilter, Layer};
 
+#[derive(Clone, Debug, Default)]
+struct ChromaShouldSample;
+
+const BUSY_NS: opentelemetry::Key = opentelemetry::Key::from_static_str("busy_ns");
+const IDLE_NS: opentelemetry::Key = opentelemetry::Key::from_static_str("idle_ns");
+
+fn is_slow(attributes: &[opentelemetry::KeyValue]) -> bool {
+    let mut nanos = 0i64;
+    for attr in attributes {
+        if attr.key == BUSY_NS || attr.key == IDLE_NS {
+            if let opentelemetry::Value::I64(ns) = attr.value {
+                nanos += ns;
+            }
+        }
+    }
+    nanos > 1_000_000
+}
+
+impl opentelemetry_sdk::trace::ShouldSample for ChromaShouldSample {
+    fn should_sample(
```

Comment:
I think this will be applied to all traces not just foyer right, is that intended if so?

Reply:
It'll run on all traces, and filter ones named "get" and "insert", which works for us today.  I'm not sure how to filter by module at this level.  I intend for it to be evaluated for every trace, but only drop get/insert-labeled traces under 1ms.

Reply:
Got it, is there anyway inside this to check that "get" and "insert" are coming from the cache module? My concern is someone introducing a span with that name and it getting filtered out here, which could be hard to detect.